cedar-policy-core/src/authorizer/partial_response.rs

```diff
@@ -333,6 +337,127 @@ impl PartialResponse {
+                        });
+                    }
+                    EntityUIDEntry::Unknown { .. } => {
+                        action = EntityUIDEntry::known(uid.clone(), None);
+                    }
+                }
+            } else {
+                return Err(ConcretizationError::ValueError {
+                    id: key.to_owned(),
+                    expected_type: "entity",
+                    given_value: val.to_owned(),
+                });
+            }
+        }
+
+        if let Some((key, val)) = mapping.get_key_value("resource") {
+            if let Ok(uid) = val.get_as_entity() {
+                match self.request.resource() {
+                    EntityUIDEntry::Known { euid, .. } => {
+                        return Err(ConcretizationError::VarConfictError {
+                            id: key.to_owned(),
+                            existing_value: euid.as_ref().clone().into(),
+                            given_value: val.clone(),
+                        });
+                    }
+                    EntityUIDEntry::Unknown { .. } => {
+                        resource = EntityUIDEntry::known(uid.clone(), None);
+                    }
+                }
+            } else {
+                return Err(ConcretizationError::ValueError {
+                    id: key.to_owned(),
+                    expected_type: "entity",
+                    given_value: val.to_owned(),
+                });
+            }
+        }
+
+        if let Some((key, val)) = mapping.get_key_value("context") {
+            if val.get_as_record().is_ok() {
+                match self.request.context() {
+                    Some(ctx) => {
+                        return Err(ConcretizationError::VarConfictError {
+                            id: key.to_owned(),
+                            existing_value: ctx.as_ref().clone(),
+                            given_value: val.clone(),
+                        });
+                    }
+                    None => {
+                        context = Some(Context::from_partial_value_unchecked(val.clone().into()));
+                    }
+                }
+            } else {
+                return Err(ConcretizationError::ValueError {
+                    id: key.to_owned(),
+                    expected_type: "record",
+                    given_value: val.to_owned(),
+                });
+            }
+        }
+
+        // We need to replace unknowns in the partial context as well
+        if let Some(ref ctx) = context {
+            if let PartialValue::Residual(residual) = ctx.as_ref() {
+                let expr = residual.substitute(mapping);
+                let extns = Extensions::all_available();
+                let eval = RestrictedEvaluator::new(&extns);
+                let partial_value =
+                    eval.partial_interpret(BorrowedRestrictedExpr::new_unchecked(&expr))?;
+                context = Some(Context::from_partial_value_unchecked(partial_value));
```

Comment:
The safety of this `_unchecked` constructor isn't immediately obvious. It depends on the earlier `Context::from_partial_value_unchecked(val.clone().into())` maintaining the invariant and then that `partial_interpret` on `Record` residual still being a `Record`.

That all seems legit, but IMO merits some additional explanation in comments. (We could also refactor to remove the need for the invariant as suggested in #541,  but that may be complicated.)